### DISCUSSION TITLE
Add custom headers to method calls

### DISCUSSION THREAD

**Discussion by ru4ert** [quarkiverse/quarkus-openapi-generator#625] [Q&A] [answered]
Is there a way to generate the interfaces with additional custom headers?

The rest client hast [custom header](https://quarkus.io/guides/rest-client-reactive#custom-headers-support) support.

You can add `quarkus.openapi-generator.codegen.spec.petstore.additional-api-type-annotations`. Is there something similar for methods?

Updating the template is one idea, but is there another way?

**Comment by ricardozanini** [comment]
Hi! Thanks for the question! There's not. You can either use custom templates or add this support in a PR.

**Reply by kouong** [reply]
Hi @ricardozanini ,

Thanks for the reply.

What do you mean exactly with adding the support in PR?

Could you please provide a simple example where custom templates are used to add custom headers to methods?

Thanks in advance! :)

**Reply by ru4ert** [reply]
@kouong https://github.com/quarkiverse/quarkus-openapi-generator/pull/635 i guess a Pull Request ^^

**Reply by kouong** [reply]
@ru4ert thanks for the reply.

by the way, actually I have a task to add a custom header to methods calls  (rest client methods generated from openapi.yaml file). Would you please have a idea how I could achieve that using custom templates? Any simple to understand example? :)

**Reply by kouong** [reply]
Ohh I saw the PR. How can the implemented feature be used ? How are the steps to follow in order to use  the feature implemented within the PR to solve my problem. Sorry I am quite new with openapi generator and looking at the PR, I am confused.. Could you please help with that?

**Reply by ru4ert** [reply]
@kouong wait until the release of this feature. Then you can add smth like:

```
quarkus.openapi-generator.codegen.spec.petstore_json.additional-request-args=@HeaderParam("x-correlation-id") String correlationId
```
to the property file and you have this x-correlation-id as a new header

**Reply by kouong** [reply]
Actually  I saw that in the documentation.

Actually when I add the property in the application.yaml  as follows and start the server I got an error (see screenshot):

quarkus
	openapi-generator:
		codegen:
			spec:
				petstore_yaml:
					additional-request-args:  @HeaderParam("x-correlation-id") String correlationId
					
![error-1](https://github.com/quarkiverse/quarkus-openapi-generator/assets/47918869/b5b83841-baa3-4cd8-8969-a907ec256334)

whenI google, Then I add single quote arround '@HeaderParam("x-correlation-id") String correlationId' (even when I just remove the @) , i got an error that the configuration key is not recognized (see screenshot) 

![error-2](https://github.com/quarkiverse/quarkus-openapi-generator/assets/47918869/fc2b714d-f564-4c35-a93a-22ebffe2dbc8)

I am quite confused why this is not working for me :(  Any pointer?

**Reply by ru4ert** [reply]
This feature is not released yet

**Reply by kouong** [reply]
@ru4ert  thanks for the keep reply! :) 

You mean the feature described in the documentation here [Add Additional Request Arguments to each API Method](https://docs.quarkiverse.io/quarkus-openapi-generator/dev/index.html#additional-request-args) is not yet released?

**Reply by ru4ert** [reply]
Please wait until the next release. Then read through the changelog. It was merged a couple days ago.

**Reply by kouong** [reply]
Okay, thank you very much for you support @ru4ert ! :)

**Comment by ricardozanini** [comment, marked as answer]
This issue was merged in this PR: https://github.com/quarkiverse/quarkus-openapi-generator/pull/635

@kouong I can release a minor release for you today if you need it. Just let me know.

**Reply by ru4ert** [reply]
@ricardozanini i am also waiting. a minor release would be nice :)

**Reply by kouong** [reply]
@ricardozanini Yes please.  It would be very  very nice to have it :)  We are actually cuurently need that feature  in our project :)
